src/js/models/maps/Map.js

```diff
@@ -269,18 +273,22 @@ define([
               assetCategories.setMapModel(this);
               this.set("layerCategories", assetCategories);
               this.unset("layers");
+              this.set('allLayers', assetCategories.getMapAssetsFlat());
             } else if (isNonEmptyArray(config.layers)) {
-              this.set("layers", new MapAssets(config.layers));
+              const layers = new MapAssets(config.layers);
+              this.set("layers", layers);
               this.get("layers").setMapModel(this);
               this.unset("layerCategories");
+              this.set('allLayers', layers);
```

Comment:
Why are we setting the same property to both `layers` and `allLayers`?

Reply:
I want a consistent way (regardless of categorized or uncategorized layers) to read a flat list of all of the layers. I could reuse the layers attribute, but right now the code assumes that only one of the two, layers or layerCategories will be populated

Reply:
Thanks for explaining!